dijkstra.c

```diff
@@ -0,0 +1,125 @@
+// }
+
+// static int32_t isValid(int8_t x, int8_t y) {
+//     return x >= 0 && x < GRID_WIDTH && y >= 0 && y < GRID_HEIGHT;
+// }
+
+void dijkstra_map(terrain_map_t *map, Location_t *start, dijk_map_t *dNode[GRID_HEIGHT][GRID_WIDTH], dijk_map_t *dn, struct character_s *npc)
+{   
+    int i, j;
+    static uint32_t initialized = 0;
+    heap_t h;
+
+    //allocate mem for Dnode
+    for (i = 0; i < GRID_HEIGHT; i++) {
+        for (j = 0; j < GRID_WIDTH; j++) {
+            dNode[i][j] = (dijk_map_t *)malloc(sizeof(dijk_map_t));
+        }
+    }
+
+    //initialize the dNode to simulate given grid size, and set stuff
+    //crashing here
+    if (!initialized) {
+        for (i = 0; i < GRID_HEIGHT; i++) {
+            for (j = 0; j < GRID_WIDTH; j++) {
+                // still failing (seg fault line 32)
+                dNode[i][j]->pos.x = i;
+                dNode[i][j]->pos.x = j;
+                //dNode[i][j]->pos = newLoc;
+                dNode[i][j]->visited =0;
+                //dNode[i][j]->hn = NULL;
+            }
+        }
+        initialized =1;
+    }
+    // initialize all costs to INF
+    for (i =0; i < GRID_HEIGHT; i++) {
+        for (j =0; j < GRID_WIDTH; j++) {
+            dNode[i][j]->cost = INT32_MAX;
+        }
+    }
+    // setting cost of player loco to zero
+    dNode[start->x][start->y]->cost = 0;
+
+    heap_init(&h, cost_cmp, NULL);
+
+    for (i =1; i < GRID_HEIGHT; i++) {
+        for( j =1; j < GRID_WIDTH; j++) {
+            dNode[i][j]->hn = heap_insert(&h, &dNode[i][j]);
+        }
+    }
+    // okay issue i think is that i dont need dn? and i dont even know wtf im doing with it because its not initialized in main LOL
+    while (h.size > 0) {
+        dijk_map_t *dn = (dijk_map_t *)heap_remove_min(&h);
+        if (dn->cost == INT32_MAX) {
+            break;
+        }
+    }
+    //mark currnode as visited
+    dn->visited = 1;
+    int dx, dy, neighborX, neighborY;
+
+    for(dx = -1; dx <= 1; dx++) {
+        for(dy = -1; dy <= 1; dy++){
+            // these values are somehow uninitialized in the if jumps below?
+            neighborX = dn->pos.x + dx;
+            neighborY = dn->pos.y + dy;
+
+            if(neighborX >= 0 && neighborY <GRID_HEIGHT &&
+               neighborY >= 0 && neighborY <GRID_WIDTH){
+                dijk_map_t **neighbor = &dNode[neighborX][neighborY];
+                // ik this shit wrong, but need to fix other stuff before code compiles to this line
+                // this npc->cost is why currently cost map is filled with 47
+                uint32_t newCost = dn->cost + npc->cost;
+
+                if(newCost < (*neighbor)->cost) {
+                    (*neighbor)->cost = newCost;
+                    heap_decrease_key(&h, (*neighbor)->hn, NULL);
+                }
+            }
+        }
+    }
+}
+
+
+
+
+void print_dijkstra_map(dijk_map_t *dNode[GRID_HEIGHT][GRID_WIDTH]) {
```

Comment:
again array pointers size does not need to be in the parameter, there is a reason
int main(int argc, char** argv) is not int main(char* argv[argc]) one is that argc isn't a #define block but you get the point